### ISSUE TITLE
archive-get fails with log-level-file=off in 2.11

### ISSUE THREAD

**Issue by bradnicholson** [pgbackrest/pgbackrest#701]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.11

2. PostgreSQL version: 10.7

3. Operating system/version: Debian Stretch

4. Did you install pgBackRest from source or from a package: Source

5. Please attach the following as applicable:
```
[global]
log-level-console=warn
log-level-file=off
log-level-stderr=warn
repo-s3-bucket=<REDACTED>
repo-s3-endpoint=<REDACTED>
repo-s3-key=<REDACTED>
repo-s3-key-secret=<REDACTED>
repo-s3-region=<REDACTED>
repo-type=s3
start-fast=y
backup-standby=n
repo-path=<REDACTED>
recovery-option=standby_mode=on
archive-async=y
spool-path=<REDACTED>
process-max=2
compress-level=1
[test]
db-path=<REDACTED>
db-port=<REDACTED>
[global:archive-get]
process-max=2
[global:archive-push]
process-max=2
```
   
7. Describe the issue:

`archive-get` worked with the above configuration file for versions of pgBackRest < 2.11.

`archive-get` in 2.11 fails with 

```
ERROR: [064]: local-1 process terminated unexpectedly [64]
2019-03-29 14:45:38.800 P00   INFO: archive-get command end: aborted with exception [064]
```  

If I set `log-level-file` to something other than `off`, it works.

Broken:

```
/usr/bin/pgbackrest --config=/tmp/pgbackrest.conf --stanza=test  --log-level-console=info   archive-get 0000000200000002000000D8  /tmp/0000000200000002000000D8
2019-03-29 14:47:12.078 P00   INFO: archive-get command begin 2.11: [0000000200000002000000D8, /tmp/0000000200000002000000D8] --archive-async --compress-level=1 --config=/tmp/pgbackrest.conf --log-level-console=info --log-level-file=off --log-level-stderr=warn --pg1-path=<REDACTED> --process-max=2 --repo1-path=<REDACTED> --repo1-s3-bucket=<REDACTED> --repo1-s3-endpoint=<REDACTED> --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=<REDACTED> --repo1-type=s3 --spool-path=<REDACTED> --stanza=test
ERROR: [064]: local-1 process terminated unexpectedly [64]
2019-03-29 14:47:12.183 P00   INFO: archive-get command end: aborted with exception [064]
```

Working:

```
/usr/bin/pgbackrest --config=/tmp/pgbackrest.conf --stanza=test   --log-level-console=info --log-level-file=info   archive-get 0000000200000002000000D8  /tmp/0000000200000002000000D8
2019-03-29 14:47:16.444 P00   INFO: archive-get command begin 2.11: [0000000200000002000000D8, /tmp/0000000200000002000000D8] --archive-async --compress-level=1 --config=/tmp/pgbackrest.conf --log-level-console=info --log-level-file=info --log-level-stderr=warn --pg1-path=<REDACTED> --process-max=2 --repo1-path=<REDACTED> --repo1-s3-bucket=<REDACTED> --repo1-s3-endpoint=<REDACTED> --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=<REDACTED> --repo1-type=s3 --spool-path=<REDACTED> --stanza=test
2019-03-29 14:47:16.883 P00   INFO: found 0000000200000002000000D8 in the archive
2019-03-29 14:47:16.888 P00   INFO: archive-get command end: completed successfully (445ms)
```

Let me know if you need more details.  I have the trace of the failed call.  It's pretty big, and I suspect it will be easy to reproduce and not needed.

**Comment by dwsteele** [comment]
As you suspected, it was pretty easy to reproduce.  I get this error though:
```
P00  ERROR: [064]: local-1 process terminated unexpectedly [64]: ERROR: [064]: unable to write to local-1: [9] Bad file descriptor
```
Which is certainly a lot more descriptive.  Are you not seeing that?

**Comment by bradnicholson** [comment]
I don't see the bad file descriptor - just 

`ERROR: [064]: local-1 process terminated unexpectedly [64]`

**Comment by dwsteele** [comment]
Fixed at 21c83eea59f5.

That was oh so much fun to track down, but I did manage to squash it in the end.  See the commit message for details.
